docs/Available-Symbols-Generators.md

```diff
@@ -548,12 +549,68 @@ This sample shows how to change the replacement value based on evaluating condit
             "value": "product"
           }
         ],
-        "separator": "/"
+        "separator": "/",
+        "removeEmptyValues": true
       }
     }
   }
 ```
 This sample will rename folder called `Api` into `Source/Api/Microsoft/Visual Studio`. Notice that File API will automatically change `/` into `\` on Windows.
 
+<a id="multichoice-join-sample"></a>Joining [multi-choice symbol](Reference-for-template.json.md#multichoice-symbols-specifics) values:
+
+`template.json`:
+```
+  "symbols": {
+    "Framework": {
+      "type": "parameter",
+      "description": "The target framework for the project.",
+      "datatype": "multichoice",
+      "choices": [
+        {
+          "choice": "netcoreapp3.1",
+          "description": "Target netcoreapp3.1"
+        },
+        {
+          "choice": "netstandard2.1",
+          "description": "Target netstandard2.1"
+        },
+        {
+          "choice": "netstandard2.0",
+          "description": "Target netstandard2.0"
+        }
+      ],
+      "defaultValue": "netstandard2.0|netstandard2.1"
+    },
+    "joinedRename": {
+      "type": "generated",
+      "generator": "join",
+      "replaces": "SupportedFrameworks",
+      "parameters": {
+        "symbols": [
+          {
+            "type": "ref",
+            "value": "Framework"
+          }
+        ],
+        "separator": ", ",
+        "removeEmptyValues": true,
+      }
+    }
```

Comment:
This is a good idea, however it might be a burden for author to define it for each multi-value choice. Should we "build-in" such generator to multi-value choice instead?

Reply:
Correct.
Hence there will be an implicit 'stringify' operation on the multichoice, that will have fixed separator (probably pipe). For more involved scenarios (e.g. https://github.com/dotnet/templating/issues/4572) will need to use `join` generator.

The scenarios in between (someone needing specific separator) are a bit of gray zone. I choosed not to special-case support that with shorthanded proparty, but I'm not strongly opinionated about that